package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+            this._view = {value: value, promise: null};
+          }
+        } else if(typeof value === "function") {
+          // Assume it is the View class itself, already fulfilled.
+          if(!this._view || this._view.value !== value) {
+            this._view = {value: value, promise: Promise.resolve(value)};
+          }
+        } else {
+          throw error.argInvalidType("view", ["nully", "string", "function"], typeof value);
+        }
+      },
+
+      _resetView: function() {
+        if(this !== _type) {
+          delete this._view;
+        }
+      },
+
+      /**
+       * Gets a promise for the default view class or factory, if any, or `null`.
+       *
+       * A default view exists if property {@link pentaho.type.Type#view}
+       * has a non-null value.
+       *
+       * @type Promise.<?function>
+       * @readOnly
+       * @see pentaho.type.Type#view
+       */
+      get viewClass() {
+        /*jshint laxbreak:true*/
+        var view = this._view;
+        return view
+            ? (view.promise || (view.promise = promiseUtil.require(view.value, localRequire)))
+            : Promise.resolve(null);
+      },
+      //endregion
+
+      /**
+       * Creates a subtype of this one.
+       *
+       * This method creates a subtype which does not have own instance or type constructors.
+       * The base type's instance and type constructors are used to _initialize_ instances and the type.
+       *
+       * To create a type with own constructors,
+       * extend from the base instance constructor instead,
+       * by calling its `extend` method.
+       *
+       * @param {object} [typeSpec] The new type specification.
+       * @param {object} [keyArgs] Keyword arguments.
+       *
+       * @return {pentaho.type.Type} The created subtype.
+       */
+      extendProto: function(typeSpec, keyArgs) {
+        if(!typeSpec) typeSpec = {};
+
+        var baseInstProto = this.instance;
+
+        // INSTANCE I
+        var instProto = Object.create(baseInstProto);
+
+        // TYPE
+        O.setConst(this, "_hasDescendants", true);
+
+        var type = Object.create(this);
+
+        var ka = keyArgs ? Object.create(keyArgs) : {};
+        ka.instance = instProto;
+        ka.exclude  = {instance: 1};
+
+        // NOTE: `type.constructor` is still the "base" constructor.
+        type.constructor(typeSpec, ka);
+
+        // INSTANCE II
+        instProto.extend(typeSpec && typeSpec.instance, {exclude: {type: 1}});
+
+        return type;
+      },
+
+      //region creation
+      /**
+       * Creates an instance of this type, given an instance specification.
+       *
+       * If the specified instance specification contains an inline type reference,
+       * in property `"_"`, the referenced type is used to create the instance
+       * (as long as it is a subtype of this type).
+       *
+       * If the specified instance specification does not contain an inline type reference
```

Comment:
Suggest:

If the specified instance specification does not contain an inline type reference,